File: app/src/main/java/link/couple/jin/couplelink/home/HomeImgAdapter.java
package link.couple.jin.couplelink.home;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.ImageView;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import link.couple.jin.couplelink.R;
import link.couple.jin.couplelink.utile.Log;
import link.couple.jin.couplelink.utile.Util;

/**
 * Created by jin on 2018-01-23.
 */

public class HomeImgAdapter extends PagerAdapter {
    Context context;
    ArrayList<String> imageList;
    LayoutInflater mInflater;
    ViewPager viewPager;

    public HomeImgAdapter(Context context, ArrayList<String> imageList, ViewPager viewPager) {
        this.context = context;
        this.imageList = imageList;
        this.viewPager = viewPager;
        mInflater = ((HomeActivity)context).getLayoutInflater();
    }

    @Override
    public int getCount() {
        return imageList.size();
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        ViewGroup layout = (ViewGroup) mInflater.inflate(R.layout.link_img_item, container, false);
        ViewHolder holder = new ViewHolder(layout);
        Util.loadImage(holder.linkImage,imageList.get(position));
        onClicked(holder.leftBtn,position);
        onClicked(holder.rightBtn,position);
        container.addView(layout);
        return layout;
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }

    @Override
    public void destroyItem(ViewGroup collection, int position, Object view) {
        collection.removeView((View) view);
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    public void onClicked(final View view, final int position) {
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (v.getId()) {
                    case R.id.left_btn:
                        if(position > 0)
                            viewPager.setCurrentItem(position-1);
                        break;
                    case R.id.right_btn:
                        if(position < getCount()+1)
                            viewPager.setCurrentItem(position+1);
                        break;
                }
            }
        });
    }

    class ViewHolder {
        @BindView(R.id.link_image)
        ImageView linkImage;
        @BindView(R.id.left_btn)
        ImageView leftBtn;
        @BindView(R.id.right_btn)
        ImageView rightBtn;

        public ViewHolder(View view) {
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/link/couple/jin/couplelink/home/HomeImgAdapter.kt
package link.couple.jin.couplelink.home

import android.app.Activity
import android.content.Context
import android.support.v4.view.PagerAdapter
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.FrameLayout
import android.widget.ImageView

import java.util.ArrayList

import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import link.couple.jin.couplelink.R
import link.couple.jin.couplelink.utile.Log
import link.couple.jin.couplelink.utile.Util

/**
 * Created by jin on 2018-01-23.
 */

class HomeImgAdapter(internal var context: Context, internal var imageList: ArrayList<String>, internal var viewPager: ViewPager) : PagerAdapter() {
    internal var mInflater: LayoutInflater

    init {
        mInflater = (context as Activity).layoutInflater
    }

    override fun getCount(): Int {
        return imageList.size
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val layout = mInflater.inflate(R.layout.link_img_item, container, false) as ViewGroup
        val holder = ViewHolder(layout)
        Util.loadImage(holder.linkImage, imageList[position])
        onClicked(holder.leftBtn, position)
        onClicked(holder.rightBtn, position)
        container.addView(layout)
        return layout
    }

    override fun getItemPosition(`object`: Any?): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun destroyItem(collection: ViewGroup, position: Int, view: Any) {
        collection.removeView(view as View)
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    fun onClicked(view: View?, position: Int) {
        view!!.setOnClickListener { v ->
            when (v.id) {
                R.id.left_btn -> if (position > 0)
                    viewPager.currentItem = position - 1
                R.id.right_btn -> if (position < count + 1)
                    viewPager.currentItem = position + 1
            }
        }
    }

    internal inner class ViewHolder(view: View) {
        @BindView(R.id.link_image)
        var linkImage: ImageView? = null
        @BindView(R.id.left_btn)
        var leftBtn: ImageView? = null
        @BindView(R.id.right_btn)
        var rightBtn: ImageView? = null

        init {
            ButterKnife.bind(this, view)
        }
    }
}
